lib/pavilion/system_plugins.py

```diff
@@ -0,0 +1,106 @@
+import collections
+from pavilion.variables import VariableSetManager
+from yapsy.IPlugin import IPlugin
+import logging
+import re
+
+LOGGER = logging.getLogger('pav.{}'.format(__name__))
+
+class PluginSystemError(RuntimeError):
+    pass
+
+_SYSTEM_PLUGINS = {}
+
+class SysVarDict( collections.UserDict ):
+
+    def __init__( self, want_deferred=True ):
+        global._SYSTEM_PLUGINS
+        super().__init__(_SYSTEM_PLUGINS)
+        self.want_deferred = want_deferred
+
+    def __getitem__( self, name ):
+        plugin = self.data[ name ]
+        return plugin.get()
+
+def add_system_plugin( system_plugin ):
+    name = system_plugin.name
+
+    if name not in _SYSTEM_PLUGINS:
+        _SYSTEM_PLUGINS[ name ] = system_plugin
+    elif priority > _SYSTEM_PLUGINS[name].priority:
+        _SYSTEM_PLUGINS[ name ] = system_plugin
+    elif priority == _SYSTEM_PLUGINS[name].priority:
+        raise PluginSystemError("Two plugins for the same system plugin have "
+                                "the same priority {}, {}."
+                                .format(system_plugin, _SYSTEM_PLUGINS[name]))
+
+def remove_system_plugin( system_plugin ):
+    name = system_plugin.name
+
+    if name in _SYSTEM_PLUGINS:
+        del _SYSTEM_PLUGINS[ name ]
+
+def get_system_plugin( name ):
+    if name not in _SYSTEM_PLUGINS:
+        raise PluginSystemError("Module not found: '{}'".format(name))
+
+    return _SYSTEM_PLUGINS[ name ]
+
+class SystemPlugin(IPlugin):
+
+    PRIO_DEFAULT = 0
+    PRIO_COMMON = 10
+    PRIO_USER = 20
+    global._SYSTEM_PLUGINS
```

Comment:
Not only is this a syntax error, I don't think it should be here anyway.

Reply:
How do you mean?